libraries/Microsoft.Bot.Builder.Dialogs/DialogContext.cs

```diff
@@ -0,0 +1,222 @@
+        public Stack<DialogInstance> Stack { get; set; }
+
+        /// <summary>
+        /// Creates a new DialogContext instance.
+        /// </summary>
+        /// <param name="dialogs">Parent dialog set.</param>
+        /// <param name="context">Context for the current turn of conversation with the user.</param>
+        /// <param name="stack">Current dialog stack.</param>
+        public DialogContext(DialogSet dialogs, ITurnContext context, Stack<DialogInstance> stack)
+        {
+            Dialogs = dialogs;
+            Context = context;
+            Stack = stack;
+        }
+
+        /// <summary>
+        /// Returns the cached instance of the active dialog on the top of the stack or `undefined` if the stack is empty.
+        /// </summary>
+        public DialogInstance Instance
+        {
+            get
+            {
+                if (Stack.Any())
+                {
+                    return Stack.Peek();
+                }
+                return null;
+            }
+        }
+
+        public DialogResult DialogResult
+        {
+            get
+            {
+                return new DialogResult
+                {
+                    Active = Stack.Count > 0,
+                    Result = _finalResult
+                };
+            }
+        }
+
+
+        /// <summary>
+        /// Pushes a new dialog onto the dialog stack.
+        /// </summary>
+        /// <param name="dialogId">ID of the dialog to start.</param>
+        /// <param name="dialogArgs">(Optional) additional argument(s) to pass to the dialog being started.</param>
+        public async Task Begin(string dialogId, object dialogArgs = null)
+        {
+            // Lookup dialog
+            var dialog = Dialogs.Find(dialogId);
+            if (dialog == null)
+            {
+                throw new Exception($"DialogContext.begin(): A dialog with an id of '{dialogId}' wasn't found.");
+            }
+            
+            // Push new instance onto stack. 
+            var instance = new DialogInstance
+            {
+                Id = dialogId,
+                State = new object()
+            };
+
+            Stack.Push(instance);
+
+            // Call dialogs begin() method.
+            await dialog.DialogBegin(this, dialogArgs);
+        }
+
+        /// <summary>
+        /// Helper function to simplify formatting the options for calling a prompt dialog. This helper will
+        /// construct a `PromptOptions` structure and then call[begin(context, dialogId, options)](#begin).
+        /// </summary>
+        /// <param name="dialogId">ID of the prompt to start.</param>
+        /// <param name="prompt">Initial prompt to send the user.</param>
+        /// <param name="choicesOrOptions">(Optional) array of choices to prompt the user for or additional prompt options.</param>
+            
+        public Task Prompt(string dialogId, string prompt, PromptOptions options = null)
```

Comment:
null chcking on dialogId

Reply:
explicit null checking should be everywhere now